WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/pages/PagesCardBuilder.kt

```diff
@@ -49,59 +56,68 @@ class PagesCardBuilder @Inject constructor(
-        return R.drawable.ic_pages_white_24dp
+    private fun getStatusIcon(status: String): Int? {
+        return when (status) {
+            DRAFT.status -> R.drawable.ic_draft_page_draft_dashboard_card
+            PUBLISH.status -> R.drawable.ic_published_page_dashboard_card
+            SCHEDULED.status -> R.drawable.ic_scheduled_page_dashboard_card
+            else -> null
+        }
     }
 
-    @Suppress("UNUSED_PARAMETER")
-    private fun getStatusText(status: String): UiString {
-        // implement the logic to get the correct text
-        return UiString.UiStringText("")
+    private fun getStatusText(status: String): UiString.UiStringRes? {
+        return when (status) {
+            DRAFT.status -> UiString.UiStringRes(R.string.dashboard_card_page_item_status_draft)
+            PUBLISH.status -> UiString.UiStringRes(R.string.dashboard_card_page_item_status_published)
+            SCHEDULED.status -> UiString.UiStringRes(R.string.dashboard_card_page_item_status_scheduled)
+            else -> null
+        }
     }
 
-    @Suppress("UNUSED_PARAMETER")
-    private fun getLastEditedOrScheduledTime(lastModifiedOrScheduledOn: Date): UiString {
-        // implement the logic to get the text
-        return UiString.UiStringText("")
+    private fun getLastEditedOrScheduledTime(page: PagesCardModel.PageCardModel): UiString {
+        return UiString.UiStringText(
+            when (page.status) {
+                DRAFT.status, PUBLISH.status -> dateTimeUtilsWrapper.javaDateToTimeSpan(page.lastModifiedOrScheduledOn)
+                SCHEDULED.status -> dateTimeUtilsWrapper.javaDateToTimeSpan(page.date)
+                else -> ""
+            }
+        )
     }
 
-    private fun getCreatePageCard(params: PagesCardBuilderParams): CreatNewPageItem {
+    private fun getCreatePageCard(pages: List<PageContentItem>, onFooterLinkClick: () -> Unit): CreatNewPageItem {
         // Create new page button is shown with image if there is
         // less than three pages for a user
-        val pages = params.pageCard?.pages ?: emptyList()
         return if (pages.isEmpty()) {
-            createNewPageCardWithAddPageMessage(params)
+            createNewPageCardWithAddPageMessage(onFooterLinkClick)
         } else if (pages.size < REQUIRED_PAGES_IN_CARD) {
-            createNewPageCardWithAddAnotherPageMessage(params)
+            createNewPageCardWithAddAnotherPageMessage(onFooterLinkClick)
         } else {
-            createNewPageCardWithOnlyButton(params)
+            createNewPageCardWithOnlyButton(onFooterLinkClick)
         }
     }
 
-    private fun createNewPageCardWithAddPageMessage(params: PagesCardBuilderParams): CreatNewPageItem {
+    private fun createNewPageCardWithAddPageMessage(onFooterLinkClick: () -> Unit): CreatNewPageItem {
         return CreatNewPageItem(
             label = UiString.UiStringRes(R.string.dashboard_pages_card_no_pages_create_page_button),
```

Comment:
(early np - since this is a draft) but you could statically import the namespace and drop the `UiString` prefix.

Reply:
Nice, Thanks. Done in 574b12e23c8a85386ec13c6966194ea71e5139b6